.github/scripts/trymerge.py

```diff
@@ -707,7 +707,27 @@ def last_commit(self) -> Any:
         return self.info["commits"]["nodes"][-1]["commit"]
 
     def get_merge_base(self) -> str:
-        return cast(str, self.info["baseRefOid"])
+        if self.merge_base:
+            return self.merge_base
+
+        last_commit_oid = self.last_commit()["oid"]
+        # Get the merge base using the GitHub REST API. This is the same as using
+        # git merge-base without the need to have git
+        json_data = gh_fetch_url(
+            f"https://api.github.com/repos/{self.org}/{self.project}/compare/{last_commit_oid}...{self.base_ref()}",
+            headers={"Accept": "application/vnd.github.v3+json"},
+            reader=json.load,
+        )
+
+        if "merge_base_commit" in json_data:
+            self.merge_base = str(json_data["merge_base_commit"]["sha"])
+
+        # Fallback to baseRefOid if the API call fails, i.e. rate limit. Note that baseRefOid
+        # points to the base ref associated with the PR or, in other words, the head of main
+        # when the PR is created or rebased. This is not necessarily the merge base commit,
+        # but it could serve as a fallback in most cases and it's readily available as part
+        # of the PR info
+        return self.merge_base if self.merge_base else str(self.info["baseRefOid"])
```

Comment:
Originally, I intended to not save `self.merge_base` here so that the fallback value would not become permanent.  But I guess your suggestion makes more sense because it avoid inconsistency.  I'll update the code accordingly.